mantidproject/mantid
# Cppcheck cache

**Description of work.**
Adds a cache dir, to avoid rebuilding the entire project each time. This
should save anywhere from 25-55 minutes per build (depending on which 
node this hits).

**To test:**
- Run cppcheck (I'd recommend setting `-DCPPCHECK_NUM_THREAD=<n>` first)
- Check `*build*/cppcheck` there should be a folder/file called cache
- Re-run it should be much faster

There is no associated issue.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
